library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -472,21 +481,38 @@ public int removeDownloads(long... ids) {
      * it was running, and it will no longer be accessible through the download manager.
      * If there are any downloaded files, partial or complete, they will be deleted.
      *
-     * @param ids the IDs of the batches to remove
+     * @param batchesIds the IDs of the batches to remove
      * @return the number of batches actually removed
      */
-    public int removeBatches(long... ids) {
-        if (ids == null || ids.length == 0) {
-            throw new IllegalArgumentException("called with nothing to remove. input param 'ids' can't be null");
+    public int removeBatches(long... batchesIds) {
+        if (batchesIds == null || batchesIds.length == 0) {
+            throw new IllegalArgumentException("called with nothing to remove. input param 'batchesIds' can't be null");
         }
-        ContentValues values = new ContentValues();
-        values.put(DownloadContract.Batches.COLUMN_DELETED, 1);
-        // if only one id is passed in, then include it in the uri itself.
-        // this will eliminate a full database scan in the download service.
-        if (ids.length == 1) {
-            return contentResolver.update(ContentUris.withAppendedId(downloadsUriProvider.getBatchesUri(), ids[0]), values, null, null);
+
+        setDeletingStatusFor(batchesIds);
+        return markBatchesBeDeleted(batchesIds);
+    }
+
+    private void setDeletingStatusFor(long[] batchesIds) {
+        ContentValues values = new ContentValues(1);
+        values.put(DownloadContract.Downloads.COLUMN_STATUS, DownloadStatus.DELETING);
+
+        if (batchesIds.length == 1) {
+            contentResolver.update(downloadsUriProvider.getContentUri(), values, COLUMN_BATCH_ID + "=?", new String[]{String.valueOf(batchesIds[0])});
         }
-        return contentResolver.update(downloadsUriProvider.getBatchesUri(), values, getWhereClauseForIds(ids), longArrayToStringArray(ids));
+
+        contentResolver.update(downloadsUriProvider.getContentUri(), values, getWhereClauseFor(batchesIds, COLUMN_BATCH_ID), longArrayToStringArray(batchesIds));
+    }
+
+    private int markBatchesBeDeleted(long[] batchesIds) {
+        ContentValues valuesDelete = new ContentValues(1);
+        valuesDelete.put(DownloadContract.Batches.COLUMN_DELETED, 1);
```

Comment:
They modify different Uris so cannot happen at the same time :(

I've splitted it on two to make it easier to read as they do two different things